acceptance/acceptance_test.go

```diff
@@ -229,1905 +99,1417 @@ func testWithoutSpecificBuilderRequirement(
+
+			t.Log("rebuild")
+			output = pack.SuccessfulRunWithOutput("build", appImageName, "-p", appPath)
+			cleanupRebuiltAppImageTask := containerManager.CleanupTaskForImageByName(appImageName)
+			defer cleanupRebuiltAppImageTask()
+
+			assertOutput = assert.NewOutputAssertionManager(output)
+			assertOutput.ReportsSuccessfulImageBuild(appImageName)
+			assertOutput.ReportsSelectingRunImageMirrorFromLocalConfig(localRunImageMirror)
+
+			assertMixedOutput := assert.NewMixedComponentOutputAssertionManager(output, mixedComponentMessenger)
+			cachedLaunchLayer := "simple/layers:cached-launch-layer"
+			assertMixedOutput.ReportsRestoresCachedLayer(cachedLaunchLayer)
+			assertMixedOutput.ReportsExporterReusesUnchangedLayer(cachedLaunchLayer)
+			assertMixedOutput.ReportsCacheReuse(cachedLaunchLayer)
+
+			assertAppImage.RunsMockAppWithOutput("Launch Dep Contents", "Cached Dep Contents")
+
+			t.Log("rebuild with --clear-cache")
+			output = pack.SuccessfulRunWithOutput("build", appImageName, "-p", appPath, "--clear-cache")
+
+			assert.NewOutputAssertionManager(output).ReportsSuccessfulImageBuild(appImageName)
+
+			assertMixedOutput = assert.NewMixedComponentOutputAssertionManager(output, mixedComponentMessenger)
+			assertMixedOutput.ReportsSkippingRestore()
+			assertMixedOutput.ReportsSkippingBuildpackLayerAnalysis()
+			assertMixedOutput.ReportsExporterReusesUnchangedLayer(cachedLaunchLayer)
+			assertMixedOutput.ReportsCacheCreation(cachedLaunchLayer)
+
+			if pack.Supports("inspect-image") {
+				t.Log("validating inspect-image output")
+				output := pack.SuccessfulRunWithOutput("inspect-image", appImageName)
+
+				expectedOutput := pack.TemplateFixture(
+					"inspect_image_local_output.txt",
+					map[string]interface{}{
+						"image_name":             appImageName,
+						"base_image_id":          containerManager.ImageIDForReference(stackManager.RunImageMirror()),
+						"base_image_top_layer":   containerManager.TopLayerDiffID(stackManager.RunImageMirror()),
+						"run_image_local_mirror": localRunImageMirror,
+						"run_image_mirror":       stackManager.RunImageMirror(),
+						"show_reference":         testLifecycle.ShouldShowReference(),
+						"show_processes":         testLifecycle.ShouldShowProcesses(),
+					},
+				)
+
+				assert.Equal(output, expectedOutput)
+			}
+		})
+
+		it("supports building app from a zip file", func() {
+			stackManager.EnsureDefaultStackCreated()
+			testBuilder := builderManager.EnsureComboBuilderExists()
+			pack.SetDefaultTrustedBuilder(testBuilder.Name())
+
+			appPath := filepath.Join("testdata", "mock_app.zip")
+			output := pack.SuccessfulRunWithOutput("build", appImageName, "-p", appPath)
+
+			assert.NewOutputAssertionManager(output).ReportsSuccessfulImageBuild(appImageName)
+
+			appImageCleanupTask := containerManager.CleanupTaskForImageByName(appImageName)
+			defer appImageCleanupTask()
+		})
+
+		when("builder is untrusted", func() {
```

Comment:
There is some stuff around "lifecycle image" that appears to be missing here (see https://github.com/buildpacks/pack/pull/673/files#diff-38e2f40dfce97bbc5beadcb903ffe700L723) unless it's captured elsewhere.

Reply:
Good catch, I'll add that back